### DISCUSSION TITLE
How to count items in a range?

### DISCUSSION THREAD

**Discussion by vladar** [kriszyp/lmdb-js#49] [Q&A] [answered]
I am curious what is the best way to count items in a given range? Is it just `db.getKeys({ start, end })`, iterating over those and counting? Or maybe there is a more efficient method that I am missing?

**Comment by kriszyp** [comment]
A dedicated method for this would certainly be much faster than iterating through the keys and incurring key conversion cost. I'll take a stab at putting one together.

**Reply by vladar** [reply]
Awesome! Now it got me thinking if we could get this method for `dupSort` databases (e.g. `getValues` with a range)?

**Comment by kriszyp** [comment, marked as answer]
Added getCount, getKeysCount, and getValuesCount in v1.5.2 (docs might come a little later, but it basically mirrors getRange, getKeys, and getValues).
